packages/application/src/shell.ts

```diff
@@ -264,6 +279,22 @@ export class LabShell extends Widget implements JupyterFrontEnd.IShell {
       this._onLayoutModified,
       this
     );
+
+    // Setup single-document-mode title bar
+    const titleWidget = (this._titleWidget = new Widget());
+    titleWidget.id = 'jp-TitlePanel-title';
+    this.add(titleWidget, 'title');
+    if (this._dockPanel.mode == 'multiple-document') {
+      this._titleHandler.panel.hide();
+    }
+
+    this.activeChanged.connect((sender, args) => {
+      let newValue = args.newValue as MainAreaWidget;
```

Comment:
What happens when the active widget isn't a `MainAreaWidget`? Is this a safe assumption?

Reply:
Fixing with a type guard in an upcoming commit.